var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -61,14 +61,24 @@ def mpi_dependent_options(self):
                 errors = ['incompatible variants given'] + errors
                 raise RuntimeError('\n'.join(errors))
         else:
-            compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
-            ]
+            if self.compiler.name == "clang":
+                compiler_opts = [
+                    '--with-mpi=1',
+                    '--with-cc=%s  -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # Avoid confusing PETSc config by clang: warning: argument unused during compilation
```

Comment:
`CPPFLAGS` are for the preprocessor; if Fortran files are preprocessed, they need these flags as well. You want to use `CXXFLAGS` for the C++ compiler.